frontend/packages/component-library/src/tooltip/WithTooltip.tsx

```diff
@@ -23,143 +25,165 @@ const tailLengths = {
   xs: 6,
 };
 
+export interface WithTooltipHandle {
+  hideTooltip: () => void;
+}
+
 export interface WithTooltipProps {
   children: ReactNode;
   tooltipOverlayClassName?: string;
   tooltipProps: TooltipProps;
 }
 
-const WithTooltip: React.FunctionComponent<WithTooltipProps> = ({
-  children,
-  tooltipOverlayClassName,
-  tooltipProps,
-}) => {
-  const [nodePosition, setNodePosition] = useState<HTMLElement | null>(null);
-  const [showTooltip, setShowTooltip] = useState<boolean>(false);
-  const [tooltipStyles, setTooltipStyles] = useState<React.CSSProperties>({});
-  const tooltipRef = useRef<HTMLDivElement | null>(null);
-  const hideTimeoutRef = useRef<number | null>(null);
-
-  // Define the additional event handlers
-  const handleShowTooltip = (
-    show: boolean,
-    event: React.SyntheticEvent<HTMLElement>,
-    isTooltip: boolean = false,
-  ) => {
-    setShowTooltip(show);
-    if (hideTimeoutRef.current) {
-      clearTimeout(hideTimeoutRef.current);
-      hideTimeoutRef.current = null;
-    }
-    if (!isTooltip) {
-      setNodePosition(show ? (event.target as HTMLElement) : null);
-    }
-  };
-
-  const handleHideTooltip = () => {
-    hideTimeoutRef.current = window.setTimeout(() => {
-      setShowTooltip(false);
-      setNodePosition(null);
-    }, 100); // Allows for small but visible close delay
-  };
-
-  const tailLength = tailLengths[tooltipProps.size || 'm'];
-
-  const updateTooltipStyles = useCallback(
-    () =>
-      updatePositionedElementStyles({
-        nodePosition,
-        positionedElementRef: tooltipRef,
-        direction: tooltipProps.direction,
-        setPositionedElementStyles: setTooltipStyles,
-        tailOffset,
-        tailLength,
-      }),
-    [nodePosition, tailLength, tooltipProps.direction],
-  );
-
-  // Effect to update tooltip styles when the tooltip is shown
-  useEffect(() => {
-    const updateTooltipPositionIfShown = () => {
-      if (showTooltip) {
-        updateTooltipStyles();
+const WithTooltip = forwardRef<WithTooltipHandle, WithTooltipProps>(
+  ({children, tooltipOverlayClassName, tooltipProps}, ref) => {
+    const [nodePosition, setNodePosition] = useState<HTMLElement | null>(null);
+    const [showTooltip, setShowTooltip] = useState<boolean>(false);
+    const [tooltipStyles, setTooltipStyles] = useState<React.CSSProperties>({});
+    const tooltipRef = useRef<HTMLDivElement | null>(null);
+    const hideTimeoutRef = useRef<number | null>(null);
+    const suppressNextFocusRef = useRef(false);
```

Comment:
Double checking my understanding - is the idea here that if we've programmatically hidden the tooltip, the next time we focus on the wrapped element, we don't want to show the tooltip again?

Reply:
yes - in the case of the settings button, the tooltip is hidden when the dropdown is displayed (`onMouseLeave` or `onBlur`). so then when the dropdown closes, we want to NOT show the tooltip again as it creates an undesirable user experience.